quickfixj-core/src/test/java/quickfix/mina/ssl/SSLCertificateTest.java

```diff
@@ -1,1154 +1,1519 @@
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import quickfix.Acceptor;
+import quickfix.ApplicationAdapter;
+import quickfix.ConfigError;
+import quickfix.DefaultMessageFactory;
+import quickfix.FixVersions;
+import quickfix.Initiator;
+import quickfix.MemoryStoreFactory;
+import quickfix.MessageFactory;
+import quickfix.MessageStoreFactory;
+import quickfix.RuntimeError;
+import quickfix.Session;
+import quickfix.SessionFactory;
+import quickfix.SessionID;
+import quickfix.SessionSettings;
+import quickfix.ThreadedSocketAcceptor;
+import quickfix.ThreadedSocketInitiator;
+import quickfix.mina.ProtocolFactory;
+import quickfix.mina.SessionConnector;
+
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.SSLEngineResult;
+import javax.net.ssl.SSLHandshakeException;
+import javax.net.ssl.SSLSession;
+import java.math.BigInteger;
+import java.security.Principal;
+import java.security.cert.Certificate;
+import java.security.cert.X509Certificate;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.logging.Level;
+import org.apache.mina.util.AvailablePortFinder;
+import org.junit.After;
+import quickfix.mina.SocksProxyServer;
+import quickfix.test.util.SSLUtil;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertSame;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(Parameterized.class)
+public class SSLCertificateTest {
+
+    private static final String LOCALHOST_ALIAS = "localhost-quickfixj";
+
+    @Parameters
+    public static List<Object[]> parameters() {
+        return Arrays.asList(new String[][]{{"TLS_RSA_WITH_AES_128_CBC_SHA", "TLSv1.2"}, {"TLS_AES_256_GCM_SHA384", "TLSv1.3"}});
+    }
+
+    // Note: To diagnose cipher suite errors, run with -Djavax.net.debug=ssl:handshake
+    private final String enabledCipherSuites;
+    private final String enabledProtocols;
+
+    public SSLCertificateTest(String enabledCipherSuites, String enabledProtocols) {
+        this.enabledCipherSuites = enabledCipherSuites;
+        this.enabledProtocols = enabledProtocols;
+    }
+
+    @Before
+    public void setUp() {
+        Map<String, String> hostAliases = new HashMap<>();
+        hostAliases.put(LOCALHOST_ALIAS, "127.0.0.1");
+
+        HostResolutionRequestInterceptor.INSTANCE.install(new MappedHostResolver(hostAliases), DefaultHostResolver.INSTANCE);
+    }
+
+    @After
+    public void tearDown() {
+        try {
+            Thread.sleep(500);
+        } catch (InterruptedException ex) {
+            Thread.currentThread().interrupt();
+            java.util.logging.Logger.getLogger(SSLCertificateTest.class.getName()).log(Level.SEVERE, null, ex);
+        }
```

Comment:
Should we add a `HostResolutionRequestInterceptor.INSTANCE.uninstall()` here or in an `@AfterClass` method?